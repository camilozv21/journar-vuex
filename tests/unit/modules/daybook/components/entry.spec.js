import { shallowMount } from "@vue/test-utils";
import Entry from "@/modules/daybook/components/Entry.vue";
import { journalState } from "../../../mock-data/test-journal-state";
import { test, suite } from "vitest";

suite("Pruebas en Entry Component", () => {

    const mockRouter = {
        push: (...args) => { mockRouter.calls.push(args); }
    };

    const wrapper = shallowMount(Entry, {
        props: {
        entry: journalState.entries[0]
        },
        global: {
        mocks: {
            $router: mockRouter
        }
        }
    });

    test("debe de hacer match con el snapshot", ({ expect }) => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    // test("debe de redireccionar al hacer click en el entry-container", ({ expect }) => {

    //     const entryContainer = wrapper.find('.entry-container')
    //     entryContainer.trigger('click')

    //     expect( mockRouter.push ).toHaveBeenCalledWith({
    //             name: 'entry', 
    //             params: {
    //                 id: journalState.entries[0].id
    //             }
    //         })

    // });

    test("pruebas en las propiedades computadas", ({ expect }) => {

        expect( wrapper.vm.day ).toBe(23)
        expect( wrapper.vm.month ).toBe('Julio')
        expect( wrapper.vm.yearDay ).toBe('2021, Viernes')

    });
});