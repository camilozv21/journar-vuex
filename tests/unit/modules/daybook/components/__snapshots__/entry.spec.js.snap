// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pruebas en Entry Component > debe de hacer match con el snapshot 1`] = `
"<div data-v-4a9e2957="" class="mb-3 pointer p-2 entry-container">
  <!-- Titulo -->
  <div data-v-4a9e2957="" class="entry-tittle d-flex"><span data-v-4a9e2957="" class="text-success fs-5 fw-bold">23</span><span data-v-4a9e2957="" class="mx-1 fs-5">Julio</span><span data-v-4a9e2957="" class="mx-2 fw-light">2021, Viernes</span></div>
  <div data-v-4a9e2957="" class="entry-description">Hola mundo desde mock data</div>
</div>"
`;

exports[`Pruebas en Entry Component debe de hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <!-- Titilo -->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">23</span><span class="mx-1 fs-5">Julio</span><span class="mx-2 fw-light">2021, Viernes</span></div>
  <div class="entry-description">Hola mundo desde mock data</div>
</div>
`;
